certbot-dns-google/certbot_dns_google/_internal/dns_google.py

```diff
@@ -82,7 +82,7 @@ def __init__(self, account_json=None, dns_api=None):
         scopes = ['https://www.googleapis.com/auth/ndev.clouddns.readwrite']
         if account_json is not None:
             try:
-                credentials = ServiceAccountCredentials.from_json_keyfile_name(account_json, scopes)
+                credentials = Credentials.from_json_keyfile_name(account_json, scopes)
```

Comment:
Looking at https://google-auth.readthedocs.io/en/master/reference/google.oauth2.service_account.html#google.oauth2.service_account.Credentials, `Credentials.from_json_keyfile_name` no longer exists.

It looks like `from_service_account_file` might be the relevant replacement. There's an example there also on how to pass the `scopes` correctly.

Once this is working, the same change should be made to the tests, to mock the new function instead of `from_json_keyfile_name`.